pkg/component/kubernetes/apiserverexposure/sni.go

```diff
@@ -148,15 +246,44 @@ func (s *sni) Deploy(ctx context.Context) error {
 		}
 	}
 
-	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, s.client, destinationRule, istio.DestinationRuleWithLocalityPreference(destinationRule, getLabels(), hostName)); err != nil {
+	var destinationMutateFn func() error
+	destinationMutateFn = istio.DestinationRuleWithLocalityPreference(destinationRule, getLabels(), hostName)
```

Comment:
Why not combine the variable declaration and the assignment (same for gateway and virtual service)?

```suggestion
	destinationMutateFn := istio.DestinationRuleWithLocalityPreference(destinationRule, getLabels(), hostName)
```